kedro-datasets/kedro_datasets/pandas/sql_dataset.py

```diff
@@ -453,3 +497,33 @@ def _load(self) -> pd.DataFrame:
 
     def _save(self, data: None) -> NoReturn:
         raise DataSetError("'save' is not supported on SQLQueryDataSet")
+
+    # For mssql only
+    def _adapt_mssql_date_params(self) -> None:
+        """We need to change the format of datetime parameters.
+        MSSQL expects datetime in the exact format %y-%m-%dT%H:%M:%S.
+        Here, we also accept plain dates.
+        `pyodbc` does not accept named parameters, they must be provided as a list."""
+
+        params = self._load_args.get("params", {})
+        if isinstance(params, dict):
+            raise DataSetError(
+                "pyodbc, which is used to connect to the MSSQL database does not "
+                "support named arguments. Please use `?` in your query and pass the "
+                "parameters as a `list`."
+            )
+        elif not isinstance(params, list):
```

Comment:
The lint test currently is failing because there is no `else` statement here.

Reply:
Alright, I will add one, thanks!